src/__tests__/index.tests.ts

```diff
@@ -202,7 +202,10 @@ describe("Public API", () => {
       ["C", "D"],
       ["D", "B"],
     ];
-    const expectedErrorMessage = "The dependency graph has a cycle at B which depends on A,D and is depended on by C";
+    const expectedErrorMessage = `A cycle has been detected including the following nodes:
```

Comment:
more test cases are needed: what happens if the graph has more than one cycle? what happens if the graph has two cycles which overlap?

Reply:
Interesting view, let me update the tests with different scenarios to capture other edge cases.